torch/csrc/jit/script/init.cpp

```diff
@@ -312,13 +311,28 @@ struct ModuleValue : public SugaredValue {
       return std::make_shared<ModuleValue>(v->module);
     } else if (Method* v = module->find_method(field)) {
       return std::make_shared<MethodValue>(shared_from_this(), *v);
-    } else if (NamedParameter* v = module->find_parameter(field)) {
+    } else if (NamedInput* v = module->find_parameter(field)) {
       return std::make_shared<SimpleValue>(m.get_or_add_parameter(v->slot()));
+    } else if (NamedInput* v = module->find_attribute(field)) {
+      return std::make_shared<SimpleValue>(
+          m.get_or_add_attribute(v->type, v->slot()));
     }
+
     // This can also be a call to a non-script module, or a plain
     // python method. If so return this as a python value.
     py::object py_module = py::cast(module);
     if (py::object attr = py::getattr(py_module, field.c_str(), py::none())) {
+      if (py::isinstance(
```

Comment:
Attributes should be registered during the init process. Doing it lazily here does not follow the same pattern of parameters, and it prevents a user from having an attribute exist even when a method doesn't reference it.  I see trouble with this when, for instance, someone uses a non-python way of defining a Method and it can't see the attributes because they are in a Python-only state at the moment.